x/ethbridge/keeper/msg_server.go

```diff
@@ -356,3 +386,47 @@ func (srv msgServer) SetFeeInfo(goCtx context.Context, msg *types.MsgSetFeeInfo)
 
 	return &types.MsgSetFeeInfoResponse{}, nil
 }
+
+// SignProphecy relayer sign the prophecy ID and send to Sifchain after receive the burn/lock events
+func (srv msgServer) SignProphecy(goCtx context.Context, msg *types.MsgSignProphecy) (*types.MsgSignProphecyResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	logger := srv.Keeper.Logger(ctx)
+
+	cosmosSender, err := sdk.AccAddressFromBech32(msg.CosmosSender)
+	if err != nil {
+		return nil, err
+	}
+
+	account := srv.Keeper.accountKeeper.GetAccount(ctx, cosmosSender)
+	if account == nil {
+		logger.Error("account is nil.", "CosmosSender", msg.CosmosSender)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, msg.CosmosSender)
+	}
+
+	err = srv.Keeper.ProcessSignProphecy(ctx, msg)
+
+	// if error is ErrProphecyFinalized, will continue and emit event, not return error.
+	if err != nil && err != oracletypes.ErrProphecyFinalized {
+		logger.Error("keeper failed to process rescue native_token message.", errorMessageKey, err.Error())
+		return nil, err
+	}
+
+	logger.Info("sifnode received the sign prophecy message.",
+		"Message", msg)
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.CosmosSender),
+		),
+		sdk.NewEvent(
+			types.EventTypeSignProphecy,
+			sdk.NewAttribute(types.AttributeKeyCosmosSender, msg.CosmosSender),
+			sdk.NewAttribute(types.AttributeKeyNetworkDescriptor, msg.NetworkDescriptor.String()),
+			sdk.NewAttribute(types.AttributeKeyProphecyID, string(msg.ProphecyId)),
```

Comment:
We should add an attribute that is the cosmos nonce of that message that just got signed.

Reply:
will add this attribute